build.gradle

```diff
@@ -3,26 +3,329 @@ buildscript {
+    group = 'Clean'
+    buildFile = file('examples/build.gradle')
+    tasks = ['clean']
+}
+
+task cleanLocalMavenRepos(type:Delete) {
+    description = 'Remove any Realm artifacts from the local Maven repositories'
+    group = 'Clean'
+    dependsOn checkEnvironment
+    delete "${System.env.ANDROID_HOME}/extras/android/m2repository/io/realm"
+    delete "${System.env.HOME}/.m2/repository/io/realm"
+}
+
+task clean {
+    description = 'Perform all the other clean tasks'
+    group = 'Clean'
+    dependsOn cleanRealm
+    dependsOn cleanGradlePlugin
+    dependsOn cleanExamples
+    dependsOn cleanLocalMavenRepos
+}
+
+task uploadDistributionPackage(type: Exec) {
+    group = 'Release'
+    description = 'Upload the distribution package to S3'
+    dependsOn distributionPackage
+    commandLine 's3cmd', 'put', "${buildDir}/outputs/distribution/realm-java-${currentVersion}.zip", 's3://static.realm.io/downloads/java/'
+}
+
+task createEmptyFile(type: Exec) {
+    group = 'Release'
+    description = 'Create an empty file that will serve as a link on S3'
+    dependsOn uploadDistributionPackage
+    commandLine 'touch', 'latest'
+}
+
+['java', 'android'].each() { link ->
+    task "upload${link.capitalize()}LatestLink"(type: Exec) {
+        group = 'Release'
+        description = "Update the link to the latest version for ${link.capitalize()}"
+        dependsOn createEmptyFile
+        commandLine 's3cmd', 'put', 'latest', "--add-header=x-amz-website-redirect-location:/downloads/java/realm-java-${currentVersion}.zip", "s3://static.realm.io/downloads/${link}/latest"
+    }
+}
+
+task uploadUpdateVersion(type: Exec) {
+    group = 'Release'
+    description = 'Update the file on S3 containing the latest version'
+    ['java', 'android'].each() { link ->
+        dependsOn "upload${link.capitalize()}LatestLink"
+    }
+    commandLine 's3cmd', 'put', "${rootDir}/version.txt", 's3://static.realm.io/update/java'
+}
+
+task distribute {
+    group = 'Release'
+    description = 'Distribute release artifacts to Bintray and S3'
+    dependsOn uploadUpdateVersion
+}
+
+def commonPom = {
+    licenses {
+        license {
+            name 'The Apache Software License, Version 2.0'
+            url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+            distribution 'repo'
+        }
+    }
+    issueManagement {
+        system 'github'
+        url 'https://github.com/realm/realm-java/issues'
+    }
+    scm {
+        url 'scm:https://github.com/realm/realm-java'
+        connection 'scm:git@github.com:realm/realm-java.git'
+        developerConnection 'scm:git@github.com:realm/realm-java.git'
+    }
+}
+
+publishing {
+    publications {
+        library(MavenPublication) {
+            artifactId = 'realm-android'
+            artifact file("${rootDir}/realm/realm-library/build/outputs/aar/realm-library-release.aar")
+            artifact file("${rootDir}/realm/realm-library/build/outputs/aar/realm-android-${currentVersion}-sources.aar")
```

Comment:
Are sources output as aar's as well? Shouldn't this be Jars?
Same for the JavaDoc